src/db/opt.rs

```diff
@@ -1,57 +1,99 @@
     pub value_dir: PathBuf,
-    // TODO: docs
-    pub in_memory: bool,
+
+    /*  Usually modified options. */
+    /// When set to true, the DB would call an additional sync after writes to flush
+    /// mmap buffer over to disk to survive hard reboots.
     pub sync_writes: bool,
+    /// Sets how many versions to keep per key at most.
+    pub num_versions_to_keep: usize,
+    /// When set to true, the DB will be opened on read-only mode.
+    pub read_only: bool,
+    /// When set to true, everything is stored in memory. No value/sst files are
+    /// created. In case of a crash all data will be lost.
+    pub in_memory: bool,
 
-    // Memtable options
+    /* Fine tuning options. */
+    /// Sets the maximum size in bytes for memtable.
     pub mem_table_size: u64,
+    /// Sets the maximum size in bytes for LSM table or file in the base level.
     pub base_table_size: u64,
+    /// Sets the maximum size target for the base level.
     pub base_level_size: u64,
+    /// Sets the ratio between the maximum sizes of contiguous levels in the LSM.
     pub level_size_multiplier: usize,
+    /// Sets the ratio between the maximum sizes of table in contiguous levels.
     pub table_size_multiplier: usize,
+    /// Sets the maximum number of levels of compaction allowed in the LSM.
     pub max_levels: usize,
 
+    /// Sets the threshold used to decide whether a value is stored directly in the
+    /// LSM tree or separately in the log value files.
     pub value_threshold: usize,
+    /// Sets the maximum number of tables to keep in memory before stalling.
     pub num_memtables: usize,
 
+    /// Sets the size of any block in SSTable.
     pub block_size: usize,
+    /// Sets the false positive probability of the bloom filter in any SSTable.
     pub bloom_false_positive: f64,
 
+    /// Sets the maximum number of Level 0 tables before compaction starts.
     pub num_level_zero_tables: usize,
+    /// Sets the number of Level 0 tables that once reached causes the DB to
+    /// stall until compaction succeeds.
     pub num_level_zero_tables_stall: usize,
 
+    /// Sets the maximum size of a single value log file.
     pub value_log_file_size: u64,
+    /// Sets the maximum number of entries a value log file can hold approximately.
     pub value_log_max_entries: u32,
+
+    /// Sets the number of compaction workers to run concurrently.
+    pub num_compactors: usize,
+    /// Indicates when the db should verify checksums for SSTable blocks.
+    pub checksum_mode: opt::ChecksumVerificationMode,
 }
 
 impl Default for AgateOptions {
     fn default() -> Self {
         Self {
             dir: PathBuf::new(),
             value_dir: PathBuf::new(),
-            // memtable options
+
+            sync_writes: false,
+            num_versions_to_keep: 1,
+            read_only: false,
+            in_memory: false,
+
             mem_table_size: 64 << 20,
             base_table_size: 2 << 20,
             base_level_size: 10 << 20,
```

Comment:
The default value of base level size is not match with mem-table size, after a mem-table flushed to the level0, it is larger the base level size much more.

Reply:
>  https://github.com/GanZiheng/agatedb/blob/finish-levels/src/levels.rs#L149

For level 0, we use `mem_table_size` to determine whether it's time to do compact. And `base_level_size` is used to get the base level.